aten/src/ATen/native/cpu/MaxPoolKernel.cpp

```diff
@@ -183,15 +183,21 @@ compute_internal(
   }
 }
 
-template <typename scalar_t, typename accscalar_t>
+template <typename scalar_t, bool is_3d>
 void cpu_max_pool(
     const Tensor& output_,
     const Tensor& indices_,
     const Tensor& input_,
-    int kW, int kH,
-    int dW, int dH,
-    int padW, int padH,
-    int dilationW, int dilationH) {
+    int kW, int kH, c10::optional<int> opt_kD,
```

Comment:
use `IntArrayRef` to wrap this up ? using an additional `c10::optional<int>` is a little bit messy here.